### ISSUE TITLE
plot/plot3d: Simplify the documentation for the -A option

### ISSUE THREAD

**Issue by seisman** [GenericMappingTools/gmt#8645]
https://docs.generic-mapping-tools.org/latest/plot.html#a

The `-A` option has the following syntax:
```
-A[m|p|x|y|r|t]
```
in which `m|p` works for geographic map, `x|y` works for Cartensian plot, and `r|t` works for polar projections.

But in the source code, `m`/`y`/`r` works the same way, and `p`/`x`/`t` works the same way.
https://github.com/GenericMappingTools/gmt/blob/b34b125857afedea82fb5b4c2589298df267c774/src/psxy.c#L838-L848

So, instead of showing the `-A[m|p|x|y|r|t]`, we can simplify it to:
```
-A[x|y]
```
then the documentation can be something like:

> **x**: Draw the line along *x* first then along *y*.
> **y**: Draw the line along *y* first then along *x*.
> 
> Here, *x* and *y* have different meanings in different coordinate systems.
> - For Cartesian data, *x* and *y* means X- and Y-axis, respectively
> - For polar projection, *x* means *theta* and *y* means *r*
> - For geographic projections, *x* means *parallel* and *y* means *meridian*

I'd like to work on it if approved.

**Comment by joa-quim** [comment]
Yes, it will look better that way.

**Comment by seisman** [comment]
I'm updating the documentation of this option on the PyGMT side first, then @yvonnefroehlich found an issue at https://github.com/GenericMappingTools/pygmt/pull/3720#discussion_r1899816545, which I think is a GMT bug. Here is the CLI version just in case you want to try it:

```
gmt begin map

gmt basemap -JH10c -R0/360/-90/90 -Bafg -B+t"-Am"
echo 90 -50 | gmt plot -Sc0.2c -Gred 
gmt plot -W1p -Am << EOF
90  -50
150 50
EOF

gmt basemap -JH10c -R0/360/-90/90 -Bafg -B+t"-Ap" -Xw+1c
echo 90 -50 | gmt plot -Sc0.2c -Gred 
gmt plot -W1p -Ap << EOF
90  -50
150 50
EOF
gmt end show
```

Here are the related codes:
https://github.com/GenericMappingTools/gmt/blob/b877d7288d23766014d5fb82e876fe5039dea9b5/src/gmt_vector.c#L1607-L1653

For the `GMT_STAIR_Y` case, it should follow Y (meridian) first, then X (parallel), so we should create the temporary array for the line with constant longitude first, then another line with consistent latitude. However, the source code is doing it in the wrong way. The fix is straightforward. 

@joa-quim @Esteban82 Could you please verify if it's indeed a bug, since the codes have been here for more than ten years and nobody complains about it.

**Comment by Esteban82** [comment]
Yes @seisman, it is a bug.

![map](https://github.com/user-attachments/assets/fb106546-2a5b-49ef-b118-c645ec6f15e8)

**Comment by joa-quim** [comment]
Indeed, it's a bug.

**Comment by seisman** [comment]
Test `sample1d/sample.sh` started to fail after #8648.

Here is the comparison between the baseline image and the new image.

| Baseline | Generated |
|---|---|
|![sample](https://github.com/user-attachments/assets/27bf8c9e-b35f-4722-9711-d86fd540b728)|![sample](https://github.com/user-attachments/assets/2f6c2abe-1716-434a-947e-930dc3c829d8)|

The green circles are the input data, and the first point is at (-5, 74). When `-Am` (blue lines) is used, the data should be resampled following meridian (Y) first, then parallel (X). Apparently, in the baseline image, the lines for `-Am`/`-Ap` are incorrect.

In the image generated by the current master branch, the geographic one (upper panel) looks correct, but the Cartesian one (lower panel) is still wrong. Am I understanding it correctly?

**Comment by joa-quim** [comment]
Hmm, aren't you swapping what's wrong/right? With master (confirmed with a local run). "Generated" geog is right. "BLUE LINE: -Am mode" is right on master but not on Baseline. The inverse ... hold on ...

Lower panels show different behavior. The wither, at the top, are wrong but the narrower at the the bottom are correct.

**Comment by seisman** [comment]
The "Baseline" one on the left is the current `sample1d/sample.ps` file, in which both panels are wrong.

After #8648, we get the "Generated" one on the right, in which the "geographic" one is correct but the Cartesian one is wrong.

With PR #8867, we will get the following image, which is correct for both "geographic" and "Cartesian"
![image](https://github.com/user-attachments/assets/0e2c8714-a3f6-4029-ba71-061e2685bbc6)

**Comment by joa-quim** [comment]
I find the Cartesian example too complicate for me to understand what is right or wrong.

**Comment by remkos** [comment]
I agree that the cartesian example is a bit too complicated. But in any case with @seisman's #8667, both the cartesian and geographic versions are correct. Thus I approved #8667
